### ISSUE TITLE
(aws_cloudfront): [TypeScript] Type error for LoadBalancerV2Origin

### ISSUE THREAD

**Issue by rafalwolny** [aws/aws-cdk#18222]
### What is the problem?

I'm switching from CDK v1 to v2 and I'm trying to pass loadbalancer props as origin props in additional behaviors in Cloudfront distribution but it seems that I can't pass them due to type error.

### Reproduction Steps

Below important part of my code

```
import * as cdk from "aws-cdk-lib";
import { aws_certificatemanager as acm } from "aws-cdk-lib";
import { aws_cloudfront as cloudfront } from "aws-cdk-lib";
import { aws_cloudfront_origins as origins } from "aws-cdk-lib";
import { aws_route53 as route53 } from "aws-cdk-lib";
import { aws_route53_targets as targets } from "aws-cdk-lib";
import { aws_s3 as s3 } from "aws-cdk-lib";
import { aws_s3_deployment as s3deployment } from "aws-cdk-lib";
import { Construct } from "constructs";
import { aws_ecs_patterns as ecspatterns } from "aws-cdk-lib";

export class MyStack extends cdk.Stack {
    constructor(scope: Construct, id: string, props: MyStackProps) {
        super(scope, id, props);

        const TESTLoadBalancedEcsService =
            new ecspatterns.ApplicationLoadBalancedEc2Service(
                this,
                "testService",
            );

        const testLoadBalancerV2 = {
            env: TESTLoadBalancedEcsService.loadBalancer.env,
            loadBalancerCanonicalHostedZoneId: TESTLoadBalancedEcsService.loadBalancer.loadBalancerCanonicalHostedZoneId,
            loadBalancerDnsName: "example.com",
            node: TESTLoadBalancedEcsService.loadBalancer.node,
            stack: TESTLoadBalancedEcsService.loadBalancer.stack
        };
...
...

const someDistribution = new cloudfront.Distribution(
            this,
            "someDistribution",
            {
                defaultBehavior: {
                    origin: new origins.S3Origin(someBucket),
                },
                additionalBehaviors: {
                    "/some/path/*": {
                        origin: new origins.LoadBalancerV2Origin(
                            testLoadBalancerV2,                                 <--- here I'm passing the props and getting error
                            {
                                protocolPolicy:
                                    cloudfront.OriginProtocolPolicy.HTTPS_ONLY,
                            }
                        ),
                        viewerProtocolPolicy:
                            cloudfront.ViewerProtocolPolicy.ALLOW_ALL,
                    },
...
```

### What did you expect to happen?

Before migration to v2 this was working solution to pass props from loadbalancer to origin. I want to get a new solution for v2

### What actually happened?

Error message: 
```
Argument of type '{ env: cdk.ResourceEnvironment; loadBalancerCanonicalHostedZoneId: string; loadBalancerDnsName: string; node: Node; stack: cdk.Stack; }' is not assignable to parameter of type 'ILoadBalancerV2'. 
Property 'applyRemovalPolicy' is missing in type '{ env: cdk.ResourceEnvironment; loadBalancerCanonicalHostedZoneId: string; loadBalancerDnsName: string; node: Node; stack: cdk.Stack; }' but required in type 'ILoadBalancerV2'.
```

### CDK CLI Version

2.3.0

### Node.js Version

v16.13.0

### OS

Ubuntu 20.04.3

### Language

Typescript

**Comment by njlynch** [comment]
This is the relevant part of the error:

`Property 'applyRemovalPolicy' is missing in type [...] but required in type 'ILoadBalancerV2'.`

This isn't a v1 vs v2 issue, this is simply a new requirement of the `IResource` interface added in https://github.com/aws/aws-cdk/pull/17746. Tagging @skinny85 and @mskrip as an FYI, as they were involved in that PR.

You can address this by implementing a (no-op) `applyRemovalPolicy` to your `testLoadBalancerV2`:

```ts
        const testLoadBalancerV2 = {
            env: TESTLoadBalancedEcsService.loadBalancer.env,
            loadBalancerCanonicalHostedZoneId: TESTLoadBalancedEcsService.loadBalancer.loadBalancerCanonicalHostedZoneId,
            loadBalancerDnsName: "example.com",
            node: TESTLoadBalancedEcsService.loadBalancer.node,
            stack: TESTLoadBalancedEcsService.loadBalancer.stack,
            applyRemovalPolicy: () => { },  // <-- New line
        };
```

That being said, for your use case, creating the `ILoadBalancerV2` is incredibly unnecessary. :) All the `LoadBalancerV2Origin` does is create a `HttpOrigin` with the load balancer's DNS name. So you can replace the below:

```ts
                        origin: new origins.LoadBalancerV2Origin(
                            testLoadBalancerV2,
                            {
                                protocolPolicy:
                                    cloudfront.OriginProtocolPolicy.HTTPS_ONLY,
                            }
                        )
```

With this:

```ts
                        origin: new origins.HttpOrigin(
                            'example.com',
                            {
                                protocolPolicy:
                                    cloudfront.OriginProtocolPolicy.HTTPS_ONLY,
                            }
                        )
```

Hope that helps! Feel free to mention me with any follow-ups.

**Comment by rafalwolny** [comment]
Hi thank You for explaining the root of that problem. I didn't tell everything about my issue, it was little bit more complex due to the fact I created the loadbalancer in the other stack which was in the other region. In further investigation I found that cross region reference was the problem (I forgot to write about that) and that was why the interface was created initially (it was a workaround to pass the data from loadbalancer to  Cloudfront distribution).

Just FYI -> I added `applyRemovalPolicy: () => { },` to the interface and it seems to be working fine. Thank You for support.
